app/models/application_record.rb

```diff
@@ -1,10 +1,24 @@
 require "acts_as_tenant"
 
 class ApplicationRecord < ActiveRecord::Base
+  include Pundit
+
   self.abstract_class = true
 
   require 'act_as_taggable_on'
   ActiveSupport.on_load(:active_record) do
     extend ActAsTaggableOn
   end
+
+  private
+
+  def user_capabilities
+    return nil if user_context.nil?
+
+    PolicyFinder.new(self).policy.new(user_context, self).user_capabilities
```

Comment:
Not all models will expose user_capabilities.  Was a mixin considered and if so what lead to using this base class?  Just wondering.

Reply:
@mkanoor Correct me if I'm wrong but I believe the reason for putting this on the base class is to stay consistent with the solution from the approval side as well.

This is also why I made it private though, if you want to actually expose it, you should do so through the `#metadata` method and then call this.

Reply:
@gmcculloug Every class that needs to have metadata/user_capabilties needs to define a rails virtual attribute called metadata, for us that is only in the Portfolio and PortfolioItem class. The rest of the classes don't have this attribute and hence the metadata method is not called.

Reply:
Understood, but the current logic means that all the models define it, which seems odd knowing most are not properly setup, or expected, to ever use it.  And going a step further if you used a mixin you could use an `included do` block to define the virtual attribute in the mixin class instead of each app/model.

I can even see the default `metadata` method, with the changes in this PR moving into the mixin.

@bzwei @hsong-rh Is this how it is implemented in approval and was there a reason behind it that would justify keeping it this way?